tools/delete_imaging_upload.pl

```diff
@@ -0,0 +1,741 @@
+
+
+RETURNS: 
+  - backup file name.
+
+=cut
+sub getBackupFileName {
+    my($tarchiveID) = @_;
+    
+    return "imaging_upload.$tarchiveID.tar.gz";
+}
+
+=pod
+
+=head3 backupFiles($archiveLocation, $filePathsRef)
+
+Backs up all the files associated to the archive before deleting them. The backed up files will
+be stored in a C<.tar.gz> archive where all paths are relative to C</> (i.e absolute paths).
+
+INPUTS:
+  - $archiveLocation: full path of the archive associated to the upload(s) passed on the
+                      command line (computed using the C<ArchiveLocation> value in table 
+                      C<tarchive> for the given archive).
+  - $filePathsRef: reference to the array that contains the absolute paths of all files found in tables
+                   C<files>, C<files_intermediary>, C<parameter_file>, C<mri_protocol_violated_scans>
+                   C<mri_violations_log> and C<MRICandidateErrors> that are tied to the upload(s) passed
+                   on the command line.
+  - $tarchiveID: ID of the DICOM archive (in table C<tarchive>) associated to the upload(s) passed on the command line.
+                 
+=cut
+sub backupFiles {
+    my($archiveLocation, $filePathsRef, $tarchiveID) = @_;
+    
+    # Put in @files the absolutes paths of all files in tables files, files_intermediary
+    # parameter_file, mri_protocol_violated_scans, mri_violations_log and MRICandidateErrors
+    # that are tied to the upload(s). Also put in @files the archive itself (found in table tarchive).
+    my @files;
+    push(@files, map { $_->{'File'} }          @{ $filePathsRef->{'files'} });
+    push(@files, map { $_->{'File'} }          @{ $filePathsRef->{'files_intermediary'} });
+    push(@files, map { $_->{'Value'} }         @{ $filePathsRef->{'parameter_file'} });
+    push(@files, map { $_->{'minc_location'} } @{ $filePathsRef->{'mri_protocol_violated_scans'} });
+    push(@files, map { $_->{'MincFile'} }      @{ $filePathsRef->{'mri_violations_log'} });
+    push(@files, map { $_->{'MincFile'} }      @{ $filePathsRef->{'MRICandidateErrors'} });
+    push(@files, $archiveLocation);
+    
+    foreach my $f (@files) {
+        # If file does not exist (i.e database and file system are not in sync)
+        if(!-e $f) {
+            if($dieOnFileError) {
+                print STDERR "Cannot backup database files:\n";
+                print STDERR "\tFile $f is in the database but was not found in $dataDirBasepath. Aborting.\n";
+                exit $NeuroDB::ExitCodes::MISSING_FILES;
+            }
+            warn "Warning! File $f was not found in $dataDirBasepath (will not be backed up)\n";
+        } elsif(!-r $f) {
+            if($dieOnFileError) {
+                print STDERR "Cannot backup database files:\n";
+                print STDERR "\tFile $f is in database but is not readable. Aborting.\n";
+                exit $NeuroDB::ExitCodes::UNREADABLE_FILE;
+            }
```

Comment:
should probably remove the path of the file from the list of files to tar when running the script with `-ignore` option here too